azure-pipelines.yml

```diff
@@ -0,0 +1,191 @@
+
+        - task: PowerShell@2
+          condition: eq(variables['Build.SourceBranch'], 'refs/release')
+          displayName: 'Pre-release versioning - CHANGELOG.md'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $changelogContent = Get-Content -Path "$(Build.SourcesDirectory)/device_calendar/CHANGELOG.md" -Raw
+
+              Write-Host " (i) Current version: $env.currentVersion"
+              Write-Host " (i) Original CHANGELOG.md content: $changelogContent"
+
+              [string] $newChangelogContent = $changelogContent -replace $env:currentVersion, "$env:currentVersion-dev.$(Build.BuildID)"
+
+              Write-Host " (i) Updated CHANGELOG.md content: $newChangelogContent"
+
+              $newChangelogContent | Set-Content -Path "$(Build.SourcesDirectory)/device_calendar/CHANGELOG.md"
+
+        - task: CmdLine@2
+          displayName: 'Dry run publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --dry-run'
+
+        - task: CmdLine@2
+          condition: eq(variables['Build.SourceBranch'], 'refs/release')
+          displayName: 'Publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --force'
+            
+  # ----------- CD Production -----------
+  - stage: Release
+    dependsOn: Build
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/master'))
+    jobs: 
+     - job: Release
+       pool:
+        vmImage: 'windows-latest'
+       steps:
+        - task: FlutterInstall@0
+          displayName: 'Flutter install'
+          inputs:
+            channel: 'stable'
+            version: 'latest'
+
+        - task: PowerShell@2
+          displayName: 'Get version from pubspec.yaml'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $version = select-string -path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml" -pattern "(?<=version: ).*" | %{ $_.Matches[0].Value }
+              
+              Write-Host "##vso[task.setvariable variable=currentVersion]$version"
+
+        - task: PowerShell@2
+          displayName: 'Get version summary from CHANGELOG.md'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $summary = select-string -path "$(Build.SourcesDirectory)/device_calendar/CHANGELOG.md" -pattern "($([regex]::escape($env:currentVersion))).*" | %{ $_.Matches[0].Value }
+              
+              Write-Host "##vso[task.setvariable variable=versionSummary]$summary"
+
+        - task: GitHubReleasePublish@1
+          displayName: 'Tag a release to GitHub'
+          inputs:
+            githubEndpoint: 'GitHub Tagging'
+            manuallySetRepository: false
```

Comment:
Is this condition set properly? I mean it would indicate that you don't want to set repo manually and then in the line below you're setting it. I don't know enough about this task so might be comletely off

Reply:
Yup tested (as creating drafts) and it works fine.